### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by elmar8287**
- Create a class Classroom.
- Create the has-many/belongs-to relationship between Classroom and Student.
- Create a class Book.
- Create a class Rental.
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+require_relative 'rental'
+
 class Person
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = [1..100]
     @name = name
     @age = age
     @parent_permission = parent_permission
     @corrector = Corrector.new
+    @rentals = []
```
Comment (Dipeshtwis) [inline person.rb]: - [x] Getter and Setter for @entals should be present that includes `:name, :age, :rentals`.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @elmar8287, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (elmar8287) [reply person.rb]: Done

File: person.rb [hunk 2]
```diff
@@ -2,6 +2,9 @@
 require_relative 'rental'
 
 class Person
+  attr_accessor :name, :age, :rentals
+  attr_reader :id
```
Comment (Dipeshtwis) [inline person.rb]: - [x] We don't need any Getter and Setter for the `id`. Since `id` is just a variable. Please remove line 6 from here.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @elmar8287, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**

-------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (elmar8287) [reply person.rb]: OK, thanks for the notes!

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @elmar8287, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by elmar8287** [comment]
Thanks for the review! 👍

**Pull request merged**